### DISCUSSION TITLE
saleor-storefront unable to build with more than 3050 products? "Error: Max tries exeeded"

### DISCUSSION THREAD

**Discussion by gleniat** [saleor/saleor#8864] [Deployment]
Hello,
building storefront works OK with 3050 products. As soon as I add 10 more products to backend, building product pages starts to fail. I can't load product page in development nor build (export) the site anymore. The error is always the same: `Error: Max tries exeeded.`

```
info  - Collecting page data...

> Build error occurred
Error: Max tries exeeded
    at fetch (/app/.next/server/pages/_app.js:14687:21)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  type: 'Error'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! saleor-site@3.0.0-a.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the saleor-site@3.0.0-a.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

![bug](https://user-images.githubusercontent.com/70897052/147408352-eaa2e643-9858-41fb-9053-eb15cb29ea49.png)

Backend is responding fine during the build process. I can see JSON responses in real time and they look fine.

The error comes from saleor-storefront/src/@next/utils/ssr.ts ([show code](https://github.com/saleor/saleor-storefront/blob/ad676ebe6469198c73faa4bc80eeb04d2a05a930/src/%40next/utils/ssr.ts))

What does the error mean?
How to debug this?
What is causing such behavior?
When I make the total number of products 3050 or less, it builds fine again.

Thank you

**Comment by karolkielecki** [comment]
Hi! I've passed your issue to the team. However, saleor-storefront is no longer maintained. We've recently announced a new project: https://github.com/saleor/react-storefront that is going to replace saleor-storefront.

**Comment by francoiscpaphil** [comment]
hi
do you still have the error ?
Cannot solve, but we are considering Saleor for xtra large shop so would like to know limits (>1m skus)
thanks !

**Comment by gleniat** [comment]
Yes, I still have this issue.
3050 products seems like a hard limit.

I have tried it 200 times during last 2 months and it fails every time I exceed the 3050 threshold.

I also tried to use an older backend, start db from scratch, import different products, toggle debug mode. I have plenty of RAM on my hi-end build machine. Out of ideas now. I believe it is an issue with the frontend code.

"Saleor-storefront is no longer maintained. We've recently announced a new project."
I developed multiple projects during 2021 several months before you announced such thing.

:-(

**Reply by karolkielecki** [reply]
You receive timeout due to too many data retries. There is something wrong configured with your backend. Do you receive any issues from the backend in the console?

**Comment by francoiscpaphil** [comment]
thanks!

not sure it can help but Mirumee (the guys who have developed Saleor) have built the site of Magrudy's (see on their page)

https://mirumee.com/

which has 9m skus online (and apparently developed with saleor)

so maybe you can get hints with them

sorry i cannot help

**Comment by francoiscpaphil** [comment]
Hi
can you mp me
francois @ cpaphil.com
thanks !
Francois
